custom_metrics/privacy.js

```diff
@@ -1,24 +1,312 @@
+    let tcData = {
+      present: typeof window.__tcfapi == 'function',
+      data: null,
+      compliant_setup: null,
+    };
+    // description of `__tcfapi`: https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/TCFv2/IAB%20Tech%20Lab%20-%20CMP%20API%20v2.md#how-does-the-cmp-provide-the-api
+    try {
+      if (tcData.present) {
+        // based on https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/TCFv2/IAB%20Tech%20Lab%20-%20CMP%20API%20v2.md#gettcdata
+        window.__tcfapi('getTCData', 2, (result, success) => {
+          if (success) {
+            tcData.data = result;
+            tcData.compliant_setup = true;
+          } else {
+            // special case for consentmanager ('CMP settings are used that are not compliant with the IAB TCF')
+            // see warning at the top of https://help.consentmanager.net/books/cmp/page/changes-to-the-iab-cmp-framework-js-api
+            // cf. https://help.consentmanager.net/books/cmp/page/javascript-api
+            // Test site: https://www.pokellector.com/
+            window.__tcfapi('noncompliant_getTCData', 2, (result, success) => {
+              if (success) {
+                tcData.data = result;
+                tcData.compliant_setup = false;
+              }
+            });
+          }
+        });
+      }
+    } finally {
+      return tcData;
+    }
+  })(),
+
+  /**
+   * IAB US Privacy User Signal Mechanism “USP API”
+   * https://github.com/InteractiveAdvertisingBureau/USPrivacy
+   *
+   * Test site: https://www.nfl.com/
+   */
+  iab_usp: (() => {
+    let uspData = {
+      present: typeof window.__uspapi == 'function',
+      privacy_string: null,
+    };
+    try {
+      if (uspData.present) {
+        window.__uspapi('getUSPData', 1, (result, success) => {
+          if (success) {
+            uspData.privacy_string = result;
+          }
+        });
+      }
+    } finally {
+      return uspData;
+    }
+  })(),
+
+  /**
+   * Ads Transparency Spotlight Data Disclosure schema
+   * Only for top frame, can't access child frames (same-origin policy)
+   *
+   * Test site: unknown
+   */
+  ads_transparency_spotlight: (() => {
+    // Check `meta` tag cf. https://github.com/Ads-Transparency-Spotlight/documentation/blob/main/implement.md
+    meta_tag = document.querySelector('meta[name="AdsMetadata"]');
+    let ats = {
+      present: meta_tag != null,
+      ads_metadata: null,
+    };
+    if (ats.present) {
+      ats.ads_metadata = meta_tag.content;
+    }
+    return ats;
+  })(),
+
+  /**
+   * FLoC
+   *
+   * Test site: https://floc.glitch.me/
+   * Test site: https://www.pokellector.com/
+   *
+   * @todo Check function/variable accesses through string searches (wrappers cannot be used, as the metrics are only collected at the end of the test)
+   */
+  document_interestCohort: testPropertyStringInResponseBodies('document.+interestCohort'),
```

Comment:
Not sure about this pattern. `.+` could include anything so `document` and `interestCohort` could be very far apart in the script. Also, do you mean to use the literal `.` character here, since it's a regex?

I would expect that you'd only want to match instances of `document.interestCohort` and not things like `documentFOOinterestCohort`.

Reply:
The point is that we can't know for sure if there will be a call using a global name or some local variable ([example](https://github.com/fingerprintjs/fingerprintjs/blob/f61e348c741b5b904e1cdc6265098a51bff6a9b2/src/utils/browser.ts))


So this multiple keyword match should be better than just `testPropertyStringInResponseBodies('interestCohort')`.
And no, there is no literal `.`.